gtsfm/averaging/rotation/rotation_averaging_base.py

```diff
@@ -19,28 +24,58 @@ class RotationAveragingBase(metaclass=abc.ABCMeta):
 
     # ignored-abstractmethod
     @abc.abstractmethod
-    def run(self, num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Optional[Rot3]]) -> List[Optional[Rot3]]:
+    def run(
+        self,
+        num_images: int,
+        i2Ri1_dict: Dict[Tuple[int, int], Optional[Rot3]],
+        i2Ri1_priors: Dict[Tuple[int, int], Optional[PosePrior]],
+    ) -> List[Optional[Rot3]]:
         """Run the rotation averaging.
 
         Args:
             num_images: number of poses.
             i2Ri1_dict: relative rotations as dictionary (i1, i2): i2Ri1.
+            i2Ri1_priors: priors on relative rotations.
 
         Returns:
             Global rotations for each camera pose, i.e. wRi, as a list. The number of entries in the list is
                 `num_images`. The list may contain `None` where the global rotation could not be computed (either
                 underconstrained system or ill-constrained system).
         """
 
-    def create_computation_graph(self, num_images: int, i2Ri1_graph: Delayed) -> Delayed:
+    def create_computation_graph(
+        self,
+        num_images: int,
+        i2Ri1_graph: Delayed,
```

Comment:
BIG question: what is the difference between the graph and the priors? Why don't we just add the priors to the graph?

Reply:
i2Ri1_graph is coming from the 2-view plate. priors are coming from the loader data. We might have logic in the future which uses both these sources of information?

Reply:
I think graph here refers to the fact that it is a delayed object (and not the pose graph)? Its a terminology we had when we started, I wasnt a big fan.

Reply:
Yes. I think we should not use `_graph` prefix anymore, and hence I have dropped it since a long time. However, we have not done the migration of existing code.